python/lsst/analysis/ap/zooniverseCutouts.py

```diff
@@ -0,0 +1,371 @@
+        from a collection of sources.
+
+        Parameters
+        ----------
+        data : `pandas.DataFrame`
+            The DiaSources to extract cutouts for. Must contain at least these
+            fields: ``ra, dec, diaSourceId, ccd, visit, instrument``.
+        butler : `lsst.daf.butler.Butler`
+            The butler connection to use to load the data; create it with the
+            collections you wish to load images from.
+        outputPath : `str`
+            The path to write the output images to.
+        """
+        count = self.write_images(data, butler, outputPath)
+        manifest = self.make_manifest(data)
+        manifest.to_csv(os.path.join(outputPath, "manifest.csv"), index=False)
+        self.log.info("Wrote %d images to %s", count, outputPath)
+
+    @staticmethod
+    def _make_path(source, base_path):
+        """Return a URL or file path for this source.
+
+        Parameters
+        ----------
+        source : `pandas.DataFrame`
+            DataFrame containing at least a ``diaSourceId`` field.
+        base_path : `str`
+            Base URL or directory path, with no ending ``/``.
+
+        Returns
+        -------
+        path : `str`
+            Formatted URL or path.
+        """
+        return f"{base_path}/images/{int(source.loc['diaSourceId'])}.png"
+
+    def make_manifest(self, data):
+        """Return a Zooniverse manifest attaching image URLs to source ids.
+
+        Parameters
+        ----------
+        data : `pandas.DataFrame`
+            DataFrame retrieved from APDB, conntaining at least a
+            ``diaSourceId`` column.
+
+        Returns
+        -------
+        manifest : `pandas.DataFrame`
+            The formatted URL manifest for upload to Zooniverse.
+        """
+        manifest = pd.DataFrame()
+        manifest['external_id'] = data['diaSourceId']
+        manifest['location:1'] = data.apply(self._make_path, axis=1, args=(self.config.urlRoot.rstrip('/'),))
+        manifest['metadata:diaSourceId'] = data['diaSourceId']
+        return manifest
+
+    def write_images(self, data, butler, outputPath):
+        """Make the 3-part cutout images for each requested source and write
+        them to disk.
+
+        Creates a ``images/`` subdirectory in ``outputPath`` if one
+        does not already exist; images are written there as PNG files.
+
+        Parameters
+        ----------
+        data : `pandas.DataFrame`
+            The DiaSources to extract cutouts for. Must contain at least these
+            fields: ``ra, dec, diaSourceId, ccd, visit, instrument``.
+        butler : `lsst.daf.butler.Butler`
+            The butler connection to use to load the data; create it with the
+            collections you wish to load images from.
+        outputPath : `str`
+            The path to write the output images to.
+
+        Returns
+        -------
+        count, `int`
+            Number of images that were written.
```

Comment:
I'd added `count` under the assumption that I might put in some `try:except:` error handling in the writer. Without that (and I don't know what kind of errors I should be trapping for anyway!) I guess there's not really a need for a counter, I'll just use `len(data)` for now.

Reply:
Scratch the above: since I added some error handling below, I've refactored to have `write_images` return a list of the ids it processed successfully, and both `make_manifest` and the log message now use that list. That's much better; thank you!